File: Week_02/589. N叉树的前序遍历/preorder.cpp
class Solution {
public:
    vector<int> preorder(Node* root) {
        vector<int> ans;
        helper(root, ans);
        return ans;
    }

    void helper(Node* root, vector<int>& ans) {
        if(root) {
            ans.push_back(root->val);
            for(auto node : root->children) {
                helper(node, ans);
            }
        }
    }
};

File: Week_02/589. N叉树的前序遍历/preorder.java
class Solution {
    public List<Integer> preorder(Node root) {
        List<Integer> ans = new LinkedList<>();
        helper(root, ans);
        return ans;
    }

    public void helper(Node root, List<Integer> ans) {
        if(root != null){
            ans.add(root.val);
            for(Node node : root.children) {
                helper(node, ans);
            }
        }
    }
}